frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -478,6 +480,8 @@ export function PlotlyChart({
     }
     return config
     // We want to reload the plotlyConfig object whenever the element id changes
+    // TODO: Update to match React best practices
+    // eslint-disable-next-line react-compiler/react-compiler
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [
     element.id,
```

Comment:
I think many of these dependency choices are somewhat intended and it might take a bit of effort to refactor all of those. Is it just a recommendation from react compiler or would this prevent us from using it?

Reply:
Yes, I understand why we choose to diverge from them in many cases. React Compiler will skip optimizing the components in these cases:

<img width="922" alt="Screenshot 2024-11-27 at 12 23 49 PM" src="https://github.com/user-attachments/assets/39efbd24-1647-4b8b-bb2e-9a6c33924c3e">